packages/core-flows/src/stock-location/steps/update-stock-locations.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  FilterableStockLocationProps,
+  IStockLocationServiceNext,
+  UpdateStockLocationNextInput,
+} from "@medusajs/types"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+import {
+  convertItemResponseToUpdateRequest,
+  getSelectsAndRelationsFromObjectArray,
+} from "@medusajs/utils"
+
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { UpdateStockLocationInput } from "@medusajs/types"
+
+interface StepInput {
+  selector: FilterableStockLocationProps
+  update: UpdateStockLocationInput
+}
+
+export const updateStockLocationsStepId = "update-stock-locations-step"
+export const updateStockLocationsStep = createStep(
+  updateStockLocationsStepId,
+  async (input: StepInput, { container }) => {
+    const stockLocationService = container.resolve<IStockLocationServiceNext>(
+      ModuleRegistrationName.STOCK_LOCATION
+    )
+    const { selects, relations } = getSelectsAndRelationsFromObjectArray([
+      input.update,
+    ])
+
+    const dataBeforeUpdate = await stockLocationService.list(input.selector, {
+      select: selects,
+      relations,
+    })
+
+    const updatedStockLocations = await stockLocationService.update(
+      input.selector,
+      input.update
+    )
+
+    return new StepResponse(updatedStockLocations, dataBeforeUpdate)
+  },
+  async (revertInput, { container }) => {
+    if (!revertInput?.length) {
+      return
+    }
+
+    const stockLocationService = container.resolve<IStockLocationServiceNext>(
+      ModuleRegistrationName.STOCK_LOCATION
+    )
+
+    await stockLocationService.upsert(
+      revertInput.map((item) => ({
+        id: item.id,
+        name: item.name,
+        ...(item.metadata ? { metadata: item.metadata } : {}),
+        ...(item.address ? { address: item.address } : {}),
```

Comment:
todo: if item.metadata was null, it will be replaced with an empty object, not sure this is what we are expecting. here is my suggestion
```suggestion
        metadata: isDefined(item.metadata) ? item.metadata : null
        address: isDefined(item.address) ? item.address : null
```

wdyt?

Reply:
it wont override the fields unless they are part of the item, since the operator is spreading an empty object and not assigning an empty object thus the key `metadata` wont exist, same goes for address. 

On the other hand the code you wrote seems to nullify the metadata if it wasn't updated since it then wont be included.

Reply:
Oh yes you are right 😂 nevermind, i thought we were assigning and not spreading, my bad